openghg/analyse/_scenario.py

```diff
@@ -712,6 +712,8 @@ def _align_obs_footprint(self, resample_to: str = "coarsest", platform: Optional
         obs_data = obs_data.sel(time=slice(start_obs_slice, end_slice))
         footprint_data = footprint_data.sel(time=slice(start_footprint_slice, end_slice))
 
+        if obs_data is None or footprint_data is None:
```

Comment:
Here we don't get a `None` if the datasets don't contain the requried dates. We get a `Dataset` with a time dimension of size zero. We could check for `obs_data.time.size == 0` or something?

Reply:
Thanks for this suggestion, I struggled to add a test because there was never None. Updated as per your suggestion.